internal/app/orchestrator/orchestrator.go

```diff
@@ -210,6 +212,10 @@ func (o *orchestrator) GetReadOnlyCache() cache.ReadOnlyCache {
 	return o.cache.GetReadOnlyCache()
 }
 
+func (o *orchestrator) GetKeys() []string {
```

Comment:
What do you think of combining this with `GetReadOnlyCache()` to return `(cache.ReadOnlyCache, []string)`? I'd rather expose less interface functions from the Orchestrator if these two functions are meant to be used in sequence.

Reply:
This way you can also put a lock around these operations so we don't have the sync issue you mentioned above.

Reply:
Can you clarify this approach? I think there would be an issue with this because one comes from the cache and one comes from the downstream response map, so there isn't a lock that would encompass operations on both.

Reply:
Sorry I just saw this comment. The aggregated key should usually be 1:1 with the cache keys, but I see your point. Let's leave it as it. 👍